### PR TITLE
gh: enable tests

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
`gitAndTools.gh` is an alias now, can you change the commit to `gh` please?

File: pkgs/applications/version-management/git-and-tools/gh/default.nix [hunk 1]
```diff
@@ -30,8 +35,10 @@ buildGoModule rec {
     done
   '';
 
-  # fails with `unable to find git executable in PATH`
-  doCheck = false;
+  checkInputs = [ git ];
+  checkPhase = ''
+    make test
+  '';
```
Comment (drewrisinger) [inline pkgs/applications/version-management/git-and-tools/gh/default.nix:41]: Requires a custom ``checkPhase`` b/c it uses a custom ``buildPhase/installPhase``, so some of the shell functions (i.e. ``getGoDirs()`` that the ``buildGoModule.checkPhase`` expects are missing.

**Review by zowoq — APPROVED** [review]
Passes on darwin/nixos.

**Comment by Izorkin** [comment]
`gh` build error occurs after this update:
```
ok      github.com/cli/cli/pkg/cmd/issue/shared 0.053s
ok      github.com/cli/cli/pkg/cmd/issue/status 0.042s
--- FAIL: TestIssueView_tty_Preview (0.00s)
    --- FAIL: TestIssueView_tty_Preview/Open_issue_without_metadata (0.00s)
        view_test.go:263: output did not match regexp /Open.*marseilles opened about 9 years ago.*9 comments/
            > output
            ix of coins
            Open • marseilles opened about 10 years ago • 9 comments


              **bold story**


            ———————— Not showing 9 comments ————————


            Use --comments to view the full conversation
            View this issue on GitHub: https://github.com/OWNER/REPO/issues/123

    --- FAIL: TestIssueView_tty_Preview/Open_issue_with_metadata (0.00s)
        view_test.go:263: output did not match regexp /Open.*marseilles opened about 9 years ago.*9 comments/
            > output
            ix of coins
            Open • marseilles opened about 10 years ago • 9 comments
            8 😕 • 7 👀 • 6 ❤️ • 5 🎉 • 4 😄 • 3 🚀 • 2 👎 • 1 👍
            Assignees: marseilles, monaco
            Labels: one, two, three, four, five
            Projects: Project 1 (column A), Project 2 (column B), Project 3 (column C), Project 4 (Awaiting triage)
            Milestone: uluru


              **bold story**


            ———————— Not showing 9 comments ————————


            Use --comments to view the full conversation
            View this issue on GitHub: https://github.com/OWNER/REPO/issues/123

    --- FAIL: TestIssueView_tty_Preview/Open_issue_with_empty_body (0.00s)
        view_test.go:263: output did not match regexp /Open.*marseilles opened about 9 years ago.*9 comments/
            > output
            ix of coins
            Open • marseilles opened about 10 years ago • 9 comments
            Labels: tarot


              No description provided


            ———————— Not showing 9 comments ————————


            Use --comments to view the full conversation
            View this issue on GitHub: https://github.com/OWNER/REPO/issues/123

    --- FAIL: TestIssueView_tty_Preview/Closed_issue (0.00s)
        view_test.go:263: output did not match regexp /Closed.*marseilles opened about 9 years ago.*9 comments/
            > output
            ix of coins
            Closed • marseilles opened about 10 years ago • 9 comments
            Labels: tarot


              **bold story**


            ———————— Not showing 9 comments ————————


            Use --comments to view the full conversation
            View this issue on GitHub: https://github.com/OWNER/REPO/issues/123

FAIL
FAIL    github.com/cli/cli/pkg/cmd/issue/view   0.058s
?       github.com/cli/cli/pkg/cmd/pr   [no test files]
ok      github.com/cli/cli/pkg/cmd/pr/checkout  0.076s
ok      github.com/cli/cli/pkg/cmd/pr/checks    0.058s

```

**Comment by zowoq** [comment]
Reverted in a7dda29d8aba00517d98d5c964dab280315c2345.

**Pull request merged**